src/app/lib/utilities/preprocessor/rules/addIdsToItems.js

```diff
@@ -0,0 +1,37 @@
+import uuid from 'uuid';
+import compose from 'ramda/src/compose';
+import pathOr from 'ramda/src/pathOr';
+
+const getGroups = jsonRaw => pathOr(null, ['content', 'groups'], jsonRaw);
+
+const addIds = groups => {
+  const newGroups = groups.map(group => {
+    const newGroup = group;
```

Comment:
Why do we need this assignment? Is the desired effect to clone the object?

Reply:
Yes, the desired effect is to clone it, so as to not make changes on the original data

Reply:
This doesn't clone the item. Here only the reference is passed. Consider this example 
```js
const items = [{ name: 'Jon' }, { name: 'Doe' }];

const clone = items;
clone[0].name = 'change this name';

console.log(items)
````
Items will still be modified.

Reply:
You could split this function to make it a bit more readable. I did a quick refactor but haven't verified it works

```js
const addIdToItem = ({ id, ...item }) => ({ ...item, id: id || uuid() });

const addIdsToGroupItems = ({ items, ...group }) => ({
  ...group,
  items: items && items.map(addIdToItem),
});

const mapGroups = groups => groups.map(addIdsToGroupItems);

```